pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,225 @@
+	errReconcileWorkspaceVerbsBinding = fmt.Errorf("reconciling fleet workspace verbs cluster role binding failed")
+)
+
+// fleetWorkspaceBindingHandler manages Bindings created for the InheritedFleetWorkspacePermissions field.
+type fleetWorkspaceBindingHandler struct {
+	crbClient rbacv1.ClusterRoleBindingController
+	crbCache  rbacv1.ClusterRoleBindingCache
+	crCache   rbacv1.ClusterRoleCache
+	grCache   mgmtcontroller.GlobalRoleCache
+	rbClient  rbacv1.RoleBindingController
+	rbCache   rbacv1.RoleBindingCache
+	fwCache   mgmtcontroller.FleetWorkspaceCache
+}
+
+func newFleetWorkspaceBindingHandler(management *config.ManagementContext) *fleetWorkspaceBindingHandler {
+	return &fleetWorkspaceBindingHandler{
+		crbClient: management.Wrangler.RBAC.ClusterRoleBinding(),
+		crbCache:  management.Wrangler.RBAC.ClusterRoleBinding().Cache(),
+		grCache:   management.Wrangler.Mgmt.GlobalRole().Cache(),
+		rbClient:  management.Wrangler.RBAC.RoleBinding(),
+		rbCache:   management.Wrangler.RBAC.RoleBinding().Cache(),
+		fwCache:   management.Wrangler.Mgmt.FleetWorkspace().Cache(),
+		crCache:   management.Wrangler.RBAC.ClusterRole().Cache(),
+	}
+}
+
+// ReconcileFleetWorkspacePermissionsBindings reconciles backing RoleBindings and ClusterRoleBindings created for granting permission
+// to fleet workspaces.
+func (h *fleetWorkspaceBindingHandler) reconcileFleetWorkspacePermissionsBindings(globalRoleBinding *v3.GlobalRoleBinding) error {
+	globalRole, err := h.grCache.Get(globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("couldn't get globalRole: %w", err)
+	}
+	var returnErr error
+	if err = h.reconcileResourceRulesBindings(globalRoleBinding, globalRole); err != nil {
+		returnErr = multierror.Append(returnErr, errReconcileResourceRulesBinding, err)
+	}
+	if err = h.reconcileWorkspaceVerbsBindings(globalRoleBinding, globalRole); err != nil {
+		returnErr = multierror.Append(returnErr, errReconcileWorkspaceVerbsBinding, err)
+	}
+
+	return returnErr
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(grb *v3.GlobalRoleBinding, gr *v3.GlobalRole) error {
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("couldn't list fleetWorkspaces: %w", err)
+	}
+
+	var returnError error
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+		desiredRB := backingRoleBinding(grb, gr, fleetWorkspace.Name)
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, wrangler.SafeConcatName(grb.Name))
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, fmt.Errorf("couldn't get RoleBinding %s : %w", wrangler.SafeConcatName(grb.Name), err))
+				continue
+			}
+			if gr.InheritedFleetWorkspacePermissions != nil && gr.InheritedFleetWorkspacePermissions.ResourceRules != nil {
+				_, err = h.crCache.Get(desiredRB.RoleRef.Name)
```

Comment:
I may misunderstand how the fleetworkspace binding handler works, but should this be the rolebinding cache?
```suggestion
				_, err = h.rbCache.Get(desiredRB.RoleRef.Name)
```

Reply:
No, this is verifying that the `ClusterRole` that this `RoleBinding` is bound to exists.